<launch>
  <!-- Arguments -->
  <arg name="world_name" default="warehouse_1" />
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="-5"/>
  <arg name="z_pos" default="0.0"/>

  <env name="GAZEBO_MODEL_PATH" value="$(find warehouse_robot)/models"/>

  <include file="$(find warehouse_robot)/launch/empty_world.launch">
    <arg name="world_name" value="$(find warehouse_robot)/worlds/$(arg world_name).world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger.urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model turtlebot3_burger -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />
</launch>